exam-01/components/checkAll.js

```diff
@@ -0,0 +1,53 @@
+class CheckAll {
+    #container;
+    #data;
+    #notifyAllCheck;
+
+    constructor(container, data, notifyAllCheck) {
+        this.#container = container;
+        this.#data = data;
+        this.#notifyAllCheck = notifyAllCheck;
+
+        this.#container.addEventListener('click', this.#onCheckAllClick.bind(this));
+    }
+
+    render() {
+        let itemsLeft = this.#data.querySelectorAll('li');
+        if (itemsLeft.length != 0) {
+            this.#container.classList.remove('d-none');
+        } else {
+            this.#container.classList.add('d-none');
+        }
+    }
+
+    #onCheckAllClick() {
+        //A Viet can you double check why this isn't working for an even number of tasks, i debugged and still can't find out why
```

Comment:
This whole function is too complex which can be simplified a lot more by leaving the job to update its completed state to the `Tasks` component